FWCore/Framework/interface/resolveMaker.h

```diff
@@ -12,28 +13,56 @@ namespace edm::detail {
                                                 std::string const& modtype,
                                                 ModuleTypeResolverBase const* resolver);
 
-  template <typename TFactory>
-  auto resolveMaker(std::string const& moduleType, ModuleTypeResolverBase const* resolver) {
-    if (resolver) {
+  // Returns a tuple of
+  // - non-owning pointer to the maker
+  // - boolean indicating if insertion to cache was successful
+  //  * true: insertion successful, or maker was found from the cache
+  //  * false: failed to insert the maker to cache
+  template <typename TFactory, typename TCache>
+  auto resolveMaker(std::string const& moduleType,
+                    ModuleTypeResolverMaker const* resolverMaker,
+                    edm::ParameterSet const& modulePSet,
+                    TCache& makerCache) -> std::tuple<typename TCache::mapped_type::element_type*, bool> {
```

Comment:
Can the return type every be `{<good value>, false}` or `{nullptr, true}`? If not, then there is no need for the `bool`.

Reply:
Good catch. The `{nullptr, false}` is returned only if the insertion of the maker to the cache had failed (so that the caller can thrown an exception). Currently `ComponentFactory` throws an exception if the insertion failed, but `Factory` does not. Maybe it would make most sense to have this `resolveMaker` to throw the exception.

Reply:
Or you can just use 'nullptr' to denote a failure.

Reply:
Ended up with returning `nullptr` for failure.